### ISSUE TITLE
Conan Search query broken

### ISSUE THREAD

**Issue by mietzen** [conan-io/conan#4951]
To help us debug your issue please explain:

- [ x ] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [ x ] I've specified the Conan version, operating system version and any tool that can be relevant.
- [ x ] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

Conan version 1.14.1, 1.12.3, 1.11.2, 1.13.1
OS: Linux (Ubuntu 16.04 / Fedora 29)
Artifactory Server Version 6.9.0

Conan search doesn't work as expected since the Upgrade of our Artifactory server:

Works:
`conan search my-product/1.0.0@usr/stable -q "build_type=Release AND os=Windows" -r releases
`
Doesn't work:
`conan search my-product/1.0.0@usr/stable -q "(build_type=Release AND os=Windows) OR os=Linux" -r releases`

`ERROR: Unknown query key: (build_type. [Remote: releases]`

And with more complex Queries I get an different error:

`ERROR: Unexpected query syntax: (os=Windows OR os=Linux OR os=any OR os=None) AND ((compiler=gcc AND compiler.version=5.4) OR (compiler="Visual Studio" AND compiler.version=14) OR compiler=any OR compiler=Any OR compiler=None) AND (arch=any OR arch=x86_64 OR arch=None) AND (arch_build=x86_64 OR arch_build=None OR arch_build=any). [Remote: releases]
`

The Errors appeared first after upgrading the Artifactory Server and the queries are still working on Artifactory Server Versions 6.5.2 and 6.8.4.

**Comment by memsharded** [comment]
Confirmed it is not working anymore in Artifactory 6.9, and it worked with Artifactory 6.3.
It is not fixed in Artifactory 6.9.1.

I have tested with Artifactory CE

**Comment by memsharded** [comment]
Reported in Artifactory: https://www.jfrog.com/jira/browse/RTFACT-18952

**Comment by mietzen** [comment]
Will this be fixed in 6.9.2? This bug is really blocking us

**Comment by memsharded** [comment]
I don't know. If possible, please try to upvote the JIRA ticket: https://www.jfrog.com/jira/browse/RTFACT-18952

**Comment by Minimonium** [comment]
Also not working with options with artifactory/bintray, but works when using on local cache.
```
$ conan search zlib/1.2.11@conan/stable -r=conan-center -q "shared=False"
ERROR: Unknown query key: shared. [Remote: conan-center]

$ conan search zlib/1.2.11@conan/stable -q "shared=False"
Existing packages for recipe zlib/1.2.11@conan/stable:
...
```

**Comment by mietzen** [comment]
Are there any news concerning this bug? The JIRA ticket seems abandoned

**Comment by memsharded** [comment]
Hi @mietzen no news at the moment, ping @markgalpin

**Comment by memsharded** [comment]
Hi @mietzen 

Sorry for the delay. We are disabling this search in the server side in #5960, and falling back to the filtering in the client side (which was the behavior that was actually being used before the upgrade to Artifactory 6.9).

This fix will be available in Conan 1.20, to be released next week. Please tell me if this is good for you, or you would need something else. Thanks!
